haystack/modeling/model/feature_extraction.py

```diff
@@ -31,45 +38,104 @@
 SPECIAL_TOKENIZER_CHARS = r"^(##|Ġ|▁)"
 
 
-def get_tokenizer(
-    pretrained_model_name_or_path: str,
-    revision: str = None,
-    use_fast: bool = True,
-    use_auth_token: Optional[Union[str, bool]] = None,
-    **kwargs,
-) -> PreTrainedTokenizer:
-    """
-    Enables loading of different Tokenizer classes with a uniform interface.
-    Right now it always returns an instance of `AutoTokenizer`.
-
-    :param pretrained_model_name_or_path:  The path of the saved pretrained model or its name (e.g. `bert-base-uncased`)
-    :param revision: The version of model to use from the HuggingFace model hub. Can be tag name, branch name, or commit hash.
-    :param use_fast: Indicate if Haystack should try to load the fast version of the tokenizer (True) or use the Python one (False). Defaults to True.
-    :param use_auth_token: The API token used to download private models from Huggingface.
-                           If this parameter is set to `True`, then the token generated when running
-                           `transformers-cli login` (stored in ~/.huggingface) will be used.
-                           Additional information can be found here
-                           https://huggingface.co/transformers/main_classes/model.html#transformers.PreTrainedModel.from_pretrained
-    :param kwargs: other kwargs to pass on to `PretrainedTokenizer.from_pretrained()`
-    :return: AutoTokenizer instance
-    """
-    model_name_or_path = str(pretrained_model_name_or_path)
+FEATURE_EXTRACTORS = {
+    **{key: AutoTokenizer for key in TOKENIZER_MAPPING_NAMES.keys()},
+    **{key: AutoFeatureExtractor for key in FEATURE_EXTRACTOR_MAPPING_NAMES.keys()},
+    # "clip": CLIPProcessor,
+}
+
+
+DEFAULT_EXTRACTION_PARAMS = {
+    AutoTokenizer: {
+        "max_length": 256,
+        "add_special_tokens": True,
+        "truncation": True,
+        "truncation_strategy": "longest_first",
+        "padding": "max_length",
+        "return_token_type_ids": True,
+    },
+    AutoFeatureExtractor: {"return_tensors": "pt"},
+}
+
+
+class FeatureExtractor:
+    def __init__(
+        self,
+        pretrained_model_name_or_path: Union[str, Path],
+        revision: str = None,
+        use_fast: bool = True,
+        use_auth_token: Optional[Union[str, bool]] = None,
+        **kwargs,
+    ):
+        """
+        Enables loading of different Feature Extractors, including Tokenizers, with a uniform interface.
+
+        Use `FeatureExtractor.extract_features()` to convert your input queries/documents/images/tables/etc...
+        into vectors that can be passed to the language model.
```

Comment:
Ok for this I'm not sure it's an improvement. This is a very internal class, the user will never do it manually. The resulting tensors will be passed over by the class that initialized this Feature Extractor. I'll apply your suggestion if you confirm that's still a better wording, otherwise we might leave it as it is?